### DISCUSSION TITLE
Parsing CODEOWNERS has no support for default section owner (Gitlab)

### DISCUSSION THREAD

**Discussion by balihb** [renovatebot/renovate#29182] [Suggest an Idea]
### Tell us more.

Gitlab supports sections in CODEOWNERS and can assign default owners for sections.
https://docs.gitlab.com/ee/user/project/codeowners/index.html#set-default-owner-for-a-section

[This section](https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/update/pr/code-owners.ts#L103C1-L112C35) must be refactored to support it:
```typescript
    // Convert CODEOWNERS file into list of matching rules
    const fileOwnerRules = codeOwnersFile
      .split(newlineRegex)
      // Remove comments
      .map((line) => line.split('#')[0])
      // Remove empty lines
      .map((line) => line.trim())
      .filter(is.nonEmptyString)
      // Extract pattern & usernames
      .map(extractOwnersFromLine);
```

**Comment by rarkins** [comment]
Converted to issue: https://github.com/renovatebot/renovate/issues/29202
